README.md

```diff
@@ -1,2 +1,78 @@
 # transactionfee-info
-Bitcoin Protocol Layer Statistics
+This project provides protocol-level statistics and insights about Bitcoin: Blocks, Transactions, SegWit and more. Implemented with a Rust backend and Hugo-based frontend.
+
+## Prerequisites
+- Nix
+- Access to a Bitcoin Core node with REST API enabled
+
+### Bitcoin Core Setup
+1. Add the following lines to your `bitcoin.conf`
+2. Restart Bitcoin Core
+
+```conf
+# Enable REST API
+rest=1
+```
+
+## Quick Start with Nix
+With Nix installed enter a nix-shell to setup and start the backend and frontend
+
+```bash
+# Enter the development environment
+nix-shell
+```
+
+### Backend
+
+```bash
+# Once in nix-shell..
+
+# Setup and start the backend
+cd backend
+# Run database migrations
+diesel setup
+diesel migration run
+# This uses the default settings of connecting to localhost:8332
+# creating a SQLite database in ./db.sqlite
+# and writing the CSV files to ./csv
+# See below for more details on the command-line arguments
+cargo run --release &
+```
+
+Note that this process could take many hours if you are starting from the genesis block!
+
+### Frontend
+
+The CSVs that are generated by the backend must be available in the `frontend/static/csv` directory for the charts to show.
+They will need to be copied to there from the default `backend/csv` directory after the backend has been run.
+
+```bash
+# Once in nix-shell..
+
+# Run the frontend
+cd frontend
+hugo server -D
+```
+
+The frontend will now be available at `http://localhost:1313`
```

Comment:
This might not always be the case. E.g. when there is already a service listening on 1313. Maybe something similar to the following is better:

```suggestion
Hugo will start serving the frontend on a webserver and provide an URL to it. 
```